### DISCUSSION TITLE
Game on Gentoo is very unstable and freezes a lot

### DISCUSSION THREAD

**Discussion by palapapa** [ppy/osu#30212] [Q&A] [answered]
I was going to report an issue, but there were too many problems to report as a single issue, and I didn't want to spam, so I decided to ask here first.

I ran the game with the official AppImage. So far, these are the problems I encountered:

1. Alt-tabbing out of the game causes it to freeze, but the audio will still play. However, losing focus and then gaining focus again doesn't freeze it, only alt-tabbing. Alt-tabbing into the game also doesn't freeze it, only when out of. It also creates a "hall of mirror" effect when it freezes like this:
![20241011125302](https://github.com/user-attachments/assets/6475978c-39a4-4074-8f8b-4e8b14aa3181)

2. Pressing any key in the game causes the keyboard and mouse to freeze. However, the screen and audio won't freeze.

3. Pressing the volume up and volume down keys on my keyboard while in game causes the screen to freeze but not the audio.

4. Whenever it freezes, I have to kill the game by using the System Monitor. Every time I do that, the game creates a weirdly named file in the same directory where the AppImage is:
![20241011130659](https://github.com/user-attachments/assets/b5f52854-4692-4cfe-9e95-b7fe8cf9d21e)

5. As I am writing this, I just noticed that the game doesn't even start anymore. All I remembered was that I tried to run it in the terminal and immediately pressed ctrl-c. After that, the game just exits the moment I run it.

## System info

Operating System: Gentoo Linux 2.15
KDE Plasma Version: 6.1.5
KDE Frameworks Version: 6.5.0
Qt Version: 6.7.2
Kernel Version: 6.6.51-gentoo-dist (64-bit)
Graphics Platform: X11
Processors: 12 × Intel® Core™ i7-8700 CPU @ 3.20GHz
Memory: 23.4 GiB of RAM
Graphics Processor: NVIDIA GeForce RTX 2070 SUPER/PCIe/SSE2
Game version is 2024.1009.1a (latest, probably, I can't confirm it now that the game doesn't even start)

## Logs

I don't know which one to attach, so here's the entire logs directory. I am pretty sure I have never exited the game normally with these logs.

[logs.zip](https://github.com/user-attachments/files/17337997/logs.zip)

**Comment by palapapa** [comment]
This turned out to not be an Osu issue, as the same happens to Steam games. In fact, the problem is that the game freezes on keyboard input. The points 1 and 3 I listed are both special cases of point 2.

**Reply by peppy** [reply, marked as answer]
welcome to linux. always assume you're at fault, not the software you're using. it will save you and us a lot of effort.
